### ISSUE TITLE
CA1062 false negative when assigning to non-nullable fields first

### ISSUE THREAD

**Issue by rdewaele** [dotnet/roslyn-analyzers#6046]
### Analyzer

**Diagnostic ID**: [CA1062](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1062): `Validate arguments of public methods`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0 and 7.0.0-preview1.22274.2 

### Describe the bug

No warning is shown when directly assigning an argument of a public constructor to a non-nullable field.
However, the warning is shown when accessing a member of said argument in the constructor.

Therefore, to my understanding, it is possible to assign the argument to the field, access a member of the field, and end up in the same situation as with directly accessing the member on the argument.

### Steps To Reproduce

Minimal reproducible example:

```
using System.Collections.ObjectModel;

[assembly: CLSCompliant(false)]

namespace warnings;

public class Warning
{
    private readonly ReadOnlyCollection<int> lst1;

    public Warning(ReadOnlyCollection<int> lst1)
    {
        this.lst1 = lst1;
        int foo = lst1.Count;
        int bar = this.lst1.Count;
    }
}
```

The csproj file that was used:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0-preview1.22274.2">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

### Expected behavior

CA1062 for lines `this.lst1 = lst1;` *and* `int foo = lst1.Count;`

### Actual behavior

CA1062 *only* for `int foo = lst1.Count;`
